src/NuGet.Core/NuGet.Build.Tasks/Common/MSBuildLogger.cs

```diff
@@ -10,53 +12,144 @@ namespace NuGet.Build
     /// <summary>
     /// TaskLoggingHelper -> ILogger
     /// </summary>
-    internal class MSBuildLogger : LegacyLoggerAdapter, Common.ILogger
+    internal class MSBuildLogger : LoggerBase, Common.ILogger
     {
         private readonly TaskLoggingHelper _taskLogging;
 
+        private delegate void LogMessageWithDetails(string subcategory, 
+            string code, 
+            string helpKeyword, 
+            string file, 
+            int lineNumber, 
+            int columnNumber, 
+            int endLineNumber, 
+            int endColumnNumber, 
+            MessageImportance importance, 
+            string message, 
+            params object[] messageArgs);
+
+        private delegate void LogErrorWithDetails(string subcategory,
+            string code,
+            string helpKeyword,
+            string file,
+            int lineNumber,
+            int columnNumber,
+            int endLineNumber,
+            int endColumnNumber,
+            string message,
+            params object[] messageArgs);
+
+        private delegate void LogMessageAsString(MessageImportance importance, 
+            string message, 
+            params object[] messageArgs);
+
+        private delegate void LogErrorAsString(string message,
+            params object[] messageArgs);
+
+
+
+
         public MSBuildLogger(TaskLoggingHelper taskLogging)
         {
             _taskLogging = taskLogging;
         }
 
-        public override void LogDebug(string data)
+        public override void Log(ILogMessage message)
         {
-            _taskLogging.LogMessage(MessageImportance.Low, data);
-        }
+            var logMessage = message as IRestoreLogMessage;
```

Comment:
Call ``DisplayMessage`` to verify that the message should be shown, this looks like it would log everything.